doc/codebase_anatomy.md

```diff
@@ -0,0 +1,54 @@
+# Codebase Anatomy
+
+This document describes the the file structure of the `kb_sdk` codebase.
+
+#### Root level
+
+* `build.xml` - main Ant build configuration file for compiling, running tests, etc.
+* `bin/` - holds the executable files generated on compilation (most importantly, `kb-sdk`)
+* `doc/` - additional documentation about this codebase
+* `Dockerfile` - the docker configuration for the container that runs the SDK
+* `entrypoint` - the entrypoint bash script that is run for the SDK docker container
+* `JAR_DEPS` and `JAR_DEPS_BIN` - indexes of java jar dependencies
+* `javacc/` - shell scripts used to generate code related to parsing KIDL specs
+* `KBaseJobService.spec` - KIDL spec for a JSON RPC job service
+* `lib` - some generic libraries that get included in SDK apps
+* `Makefile` - commands for compiling, building docker images, and initializing submodules
+* `Pipfile` and `Pipfile.lock` - python dependencies for pipenv
+* `reports/` - files generated for JaCoCo test coverage reports
+* `sdkbase/` - Docker files for the image used inside actual SDK apps
+* `src/` - the main source code for this project; see below
+* `submodules/` and `submodules_hacks` - jars and other dependencies
+* `test_scripts/` - test helpers in perl, python, and js
+
+#### Source code in `/src/java/us/kbase`
+
+* `catalog/` - Services for managing, registering, and building modules in the catalog. All code here is duplicated in the `njs_wrapper`
```

Comment:
This is still a little off IMO. I would say

`A client for the KBase catalog service compiled from the catalog service KIDL specification.`

or something similar.